### PR TITLE
Add Firebase configuration for development project

### LINKED ISSUE #9: Create and setup a dev project

### PR DESCRIPTION
**Pull request by BurningAXE**
This PR adds configuration files for Firebase pointing to a development project I created
Closed #9

### CODE REVIEW

File: pubspec.lock
Comment (pthisaiveerapandian) [inline pubspec.lock:1]: is this file required?

Comment (thisaiveerapandian) [inline pubspec.lock:1]: is this file required?

Comment (BurningAXE) [reply pubspec.lock:1]: Not really but I think it will be useful to know which exact versions went into a particular release. Any objections?

Comment (thisaiveerapandian) [reply pubspec.lock:1]: my suggestion is For version control, we can use yaml file version property.

Comment (thisaiveerapandian) [reply pubspec.lock:1]: my suggestion is For version control, we can use yaml file version property.

Comment (BurningAXE) [reply pubspec.lock:1]: Hm, I'm not sure what you mean by file version property? 
My concern is that when you define versions in pubspec.yaml like "^1.0.2" you cannot be sure of the minor version that was used. It can be 1.0.2 or 1.0.3 or 1.0.4, etc.

Comment (thisaiveerapandian) [reply pubspec.lock:1]: I mean with + we can manage the code versions like: 1.0.0+1, 1.0.0+2, 1.0.0+3

**Review by thisaiveerapandian — DISMISSED** [review]

File: lib/src/pages/register.dart
Comment (samuelkchris) [inline lib/src/pages/register.dart:1]: # 1._validateAndSave()
The _validateAndSave() method does a lot of things, including validating the form fields, creating a new user object, registering the user, and navigating to the login page if the registration is successful. It might be better to extract some of these responsibilities into separate methods to improve readability and maintainability.

# 2. form fields
The form fields are nicely laid out and include appropriate validators. However, the password confirmation validator could be improved to handle cases where the value argument is null.

File: lib/src/models/users.dart
Comment (samuelkchris) [inline lib/src/models/users.dart:1]: One suggestion would be to add some validation to the constructor to ensure that required properties such as email and uid are not null.

File: lib/src/controllers/user_controller.dart
Comment (samuelkchris) [inline lib/src/controllers/user_controller.dart:1]: It's worth noting that the setUserPrefs method doesn't return anything, which might be a problem if you want to know when the user preferences have been set.
# suggestions

### isUserLoggedIn:

Consider renaming this method to something like checkUserLoggedInStatus, as it better describes what the method does.
Consider changing the return type from Future<bool> to just Future, since the method doesn't actually return a boolean value - it returns a Future that resolves to a boolean value.
Consider adding some error handling code in case an error occurs while getting the SharedPreferences instance.
i.e

`
Future checkUserLoggedInStatus() async {
  try {
    final prefs = await SharedPreferences.getInstance();
    bool isLoggedIn = prefs.getBool(global.loggedIn) ?? false;
    return isLoggedIn;
  } catch (e) {
    print('Error getting user logged in status: $e');
    return false;
  }
}
`
### setUserPrefs:

Consider adding a parameter to this method that allows the caller to specify whether the user is logged in or not, rather than always setting the loggedIn key to true.
Consider adding some error handling code in case an error occurs while getting the SharedPreferences instance.
i.e

`
void setUserPrefs({bool isLoggedIn = true}) async {
  try {
    final prefs = await SharedPreferences.getInstance();
    prefs.setBool(global.loggedIn, isLoggedIn);
  } catch (e) {
    print('Error setting user prefs: $e');
  }
}
`

**Review by samuelkchris — DISMISSED** [review]
code looks well-structured and readable, with clear separation of concerns. Keep up the good work!

File: lib/src/pages/register.dart
Comment (BurningAXE) [reply lib/src/pages/register.dart:1]: @samuelkchris Thanks! These changes are actually from another PR that is already merged into `dev`. I wrongly pointed this PR to master, which is why they appeared as new code.
@thisaiveerapandian Tagging you FYI as you did those changes.

File: lib/src/controllers/user_controller.dart
Comment (BurningAXE) [reply lib/src/controllers/user_controller.dart:1]: @samuelkchris Good suggestions! Would you mind creating an issue and raising a PR with these changes as this code is already merged in `dev`?

Comment (samuelkchris) [reply lib/src/controllers/user_controller.dart:1]: okay let me do so
